docs/proposals/error-handling-guideline.md

```diff
@@ -0,0 +1,279 @@
+
+Current behavior and output:
+
+```
+$ oras manifest push --oci-layout /sample/images:foobar:mediatype
+Error: media type is not recognized. 
+```
+
+Suggested error message:
+
+```
+$ oras manifest push --oci-layout /sample/images:foobar:mediatype
+Error: media type is not recognized. 
+Usage: oras manifest push [flags] <name>[:<tag>[,<tag>][...]|@<digest>] <file>
+You need to specify a valid media type "mediaType" in the manifest JSON or via the "--media-type" flag.
+```
+
+#### Example 5: Attach an artifact if the given option is unknown
+
+Current behavior and output:
+
+```
+$ oras attach --artifact-type oras/test localhost:5000/command/images:foobar --distribution-spec v1.0
+Error: unknown distribution specification flag: v1.0
+```
+
+Suggested error message:
+
+```
+$ oras attach --artifact-type oras/test localhost:5000/sample/images:foobar --distribution-spec ???
+Error: unknown distribution specification flag: "v1.0". 
+Available options: v1.1-referrers-api, v1.1-referrers-tag
+```
+
+#### Example 6: When attaching, if neither file reference nor annotation is provided
+
+Current behavior and output:
+
+```
+$ oras attach --artifact-type sbom/example localhost:5000/sample/images:foobar
+Error: no blob is provided
+```
+
+Suggested error message:
+
+```
+$ oras attach --artifact-type sbom/example localhost:5000/sample/images:foobar
+Error: neither file nor annotation provided in the command
+Usage: oras attach [flags] --artifact-type=<type> <name>{:<tag>|@<digest>} <file>[:<type>] [...]
+To attach to an existing artifact, please provide files via argument or annotations via flag "--annotation". Run "oras attach -h" for more options and examples
+```
+
+#### Example 7: When pushing files, if the annotation file doesn't match the required format
+
+Current behavior and output:
+
+```
+$ oras push --annotation-file sbom.json ghcr.io/library/alpine:3.9
+Error: failed to load annotations from sbom.json: json: cannot unmarshal string into Go value of type map[string]map[string]string. Please refer to the document at https://oras.land/docs/how_to_guides/manifest_annotations
+```
+
+Suggested error message:
+
+```
+$ oras push --annotation-file annotation.json ghcr.io/library/alpine:3.9
+Error: invalid annotation json file: failed to load annotations from annotation.json.
+Annotation file doesn't match the required format. Please refer to the document at https://oras.land/docs/how_to_guides/manifest_annotations
+```
+
+#### Example 8: When pushing files, if the annotation value doesn't match the required syntax
+
+Current behavior and output:
+
+```
+$ oras push --annotation "key:value" ghcr.io/library/alpine:3.9
+Error: missing key in `--annotation` flag: key:value
+```
+
+Suggested error message:
+
+```
+$ oras push --annotation "key:value" ghcr.io/library/alpine:3.9
+Error: annotation value doesn't match the required format.
+Try `oras push --annotation "key=value" ghcr.io/library/alpine:3.9`  
```

Comment:
It's impossible to restore the full input command, the best we can do provide a valid annotation value, like: (
also, simple usage need to be provided)
```
$ oras push --annotation "key:value" ghcr.io/library/alpine:3.9
Error: annotation value doesn't match the required format.
Usage: oras push [flags] <name>[:<tag>[,<tag>][...]] <file>[:<type>] [...]
Try `--annotation "key=value"`  
```

Reply:
It is not "impossible" but it is not trivial to do so.

Reply:
Updated. Print the correct usage only in the prompt information is enough. I am fine with this format.